### PR TITLE
#33 Remove endpoint for getting a user's profile

### LINKED ISSUE #33: Remove endpoint for getting a user's profile
**Related Issues**
Project: #19 
Depends on: #9 

**Description**
The existing endpoint isn't very RESTful and it'll probably need to be replaced with the "get all users" functionality introduced in #17 .  So once #9 is completed, we'll use that for getting a user's profile.

Right now, we make a call to that endpoint to determine whether they're logged in [here](https://github.com/timmyichen/dev-directory/blob/760bb5e233473482d4f80f7bba92e37d328bd57b/pages/index.tsx#L56).  Once we remove the current route, we'll need to update this to use the route created in #9, which will require extracting the user's ID from the jwt auth token stored as a cookie.

The task is:
* Remove the existing `/api/users/` API endpoint
* Update the endpoint called in `index.tsx` to the new endpoint created in #9, using the cookie jwt auth token to determine the currently-logged-in-user's ID.

**Acceptance Criteria**
* Existing `/api/users/` endpoint is removed
* The app can still detect whether someone is logged in or not

### PR DESCRIPTION
**Pull request by windyScripts**
Closes #33 

# Description

Please include a summary of the changes. Please also include relevant motivation and context. List any dependencies that are required for this change.

- Updated API call end-point for get user
- Removed redundant route
- Added user id extraction using JWT as part of the get user call.
- Added universal-cookie as a dependency

## Type of change

Please remove all except for everything applicable, and then remove this line.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] Changes have new/updated automated tests, if applicable
- [x] Changes have new/updated docs, if applicable
- [x] I have performed a self-review of my own code
- [x] I have added comments on any new, hard-to-understand code
- [x] Changes have been manually tested
- [x] Changes generate no new errors or warnings

### CODE REVIEW

**Review by timmyichen — APPROVED** [review]
Thanks for taking this on! I realized much too late that we should also be removing expired tokens, but that's outside the scope of this ticket - this will be enough to unblock the Pagination project! I'll add a ticket after this one is merged to invalidate expired tokens.

File: pages/index.tsx
Comment (Caleb-Cohen) [inline pages/index.tsx:54]: How does this work?

In my local, my JWT is: `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoxLCJpYXQiOjE2Nzk3NzE0NjUsImV4cCI6MTY4MDM3NjI2NX0.aXXWWhweVc2R_mC8KtY9hqIGP8HlQQN4Ug5vv9OhqP4`

Which is decoded to

```JS

{
  "user_id": 1,
  "iat": 1679771465,
  "exp": 1680376265
}

```
How does whatever is issuing the JWT know my user_id?

**Review by Caleb-Cohen — APPROVED** [review]
Left a question, but LGTM!

File: pages/index.tsx
Comment (devonzara) [reply pages/index.tsx:54]: The token is created when you login and it is sent to you as a cookie. Subsequent requests will include that cookie, thus letting the server know who you are.

During login (`/api/auth/login`), a `User` is found or created from your Discord user id:
https://github.com/timmyichen/dev-directory/blob/main/server/api/auth/auth.controller.ts#L37
https://github.com/timmyichen/dev-directory/blob/main/server/lib/auth.ts#L51-L59

That `User` (which includes your local user `id`) is then passed to a function that generates the JWT token:
https://github.com/timmyichen/dev-directory/blob/main/server/api/auth/auth.controller.ts#L38
https://github.com/timmyichen/dev-directory/blob/main/server/lib/auth.ts#L14-L18

Finally, that JWT token is then set as a cookie in the response:
https://github.com/timmyichen/dev-directory/blob/main/server/api/auth/auth.controller.ts#L40

File: pages/index.tsx
Comment (devonzara) [inline pages/index.tsx:49]: What's the benefit of using `universal-cookie` here instead of `req?.cookies[AUTH_COOKIE_NAME]`?

Doing a quick search, I don't see `universal-cookie` being used anywhere else... could we possibly remove it as a dependency entirely? @timmyichen

File: server/api/auth/auth.test.ts
Comment (devonzara) [inline server/api/auth/auth.test.ts:1]: nit: To keep commits small and file changes relevant to the issue, it's probably best to not include unrelated style fixes from other files.

File: server/lib/db.ts
Comment (devonzara) [inline server/lib/db.ts:1]: nit: Same as above, unrelated style fixes

File: pages/index.tsx
Comment (timmyichen) [reply pages/index.tsx:49]: Needs to work on the client too!

Comment (devonzara) [reply pages/index.tsx:49]: Fair, I suppose until the CookieStore API becomes stable and common in most browsers, a package is probably the easiest solution. 👍🏼

**Pull request merged**
